packages/app/features/checks/utils/checkUtils.ts

```diff
@@ -0,0 +1,33 @@
+import { generatePrivateKey, privateKeyToAccount } from 'viem/accounts'
+import type { EphemeralKeyPair } from 'app/features/checks/types'
+
+/**
+ * Generates a URL for claiming a check based on the sender's ID and an ephemeral key pair.
+ * The URL contains an encoded payload comprising the sender's ID, the ephemeral private key,
+ * and the ephemeral address, separated by colons and encoded for URL compatibility.
+ *
+ * @param {string} senderId - The sender's /send account id.
+ * @param {EphemeralKeyPair} ephemeralKeyPair - An object containing the ephemeral private key and address.
+ * @returns {string} The generated URL for claiming the check.
+ */
+export const generateCheckUrl = (senderId: string, ephemeralKeyPair: EphemeralKeyPair): string => {
+  const encodedPayload = encodeURIComponent(
+    `${senderId}:${ephemeralKeyPair.ephemeralPrivkey}:${ephemeralKeyPair.ephemeralAddress}`
+  )
+  return `/check/claim/${encodedPayload}`
```

Comment:
Let's experiement with encoding the keypair in the fragment. We don't want it sent to the server, but we do want it to be part of the URL. 

```suggestion
  return `/check/claim#${encodedPayload}`
```